components/binary_sensor/xpt2046.rst

```diff
@@ -0,0 +1,265 @@
+XPT2046 Touch Screen Controller
+==================================
+
+.. seo::
+    :description: Instructions for setting up XPT2046 touchscreen controller with ESPHome
+    :image: xpt2046.jpg
+    :keywords: XPT2046
+
+.. _xpt2046-component:
+
+Component/Hub
+-------------
+
+The ``xpt2046`` component allows using the touchscreen controllers
+based on the XPT2046 chip
+(`datasheet <https://datasheetspdf.com/pdf-file/746665/XPTEK/XPT2046/1>`__,
+`AZ-Delivery`_) with ESPHome. Many cheap LCD displays contain this controller.
+The :ref:`SPI <spi>` is required to be set up in your configuration for this sensor to work.
+
+.. figure:: images/xpt2046-full.jpg
+    :align: center
+    :width: 50.0%
+
+    XPT2046 Touch Screen Controller
+
+.. _AZ-Delivery: https://www.az-delivery.de/en/products/2-4-tft-lcd-touch-display
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    xpt2046:
+      id: touchscreen
+      cs_pin: 17
+      irq_pin: 16
+      update_interval: 50ms
+      report_interval: 1s
+      threshold: 400
+      dimension_x: 240
+      dimension_y: 320
+      calibration_x_min: 3860
+      calibration_x_max: 280
+      calibration_y_min: 340
+      calibration_y_max: 3860
+      swap_x_y: False
+
+    binary_sensor:
+      - platform: xpt2046
+        xpt2046_id: touchscreen
+        id: touch_key0
+        x_min: 80
+        x_max: 160
+        y_min: 106
+        y_max: 212
+        on_state:
+          - lambda: 'ESP_LOGI("main", "key0: %s", (x ? "touch" : "release"));'
+  
+
+Configuration variables:
+------------------------
+The configuration is made up of two parts: The touchscreen component, and optional individual binary sensors.
+
+Base Configuration:
+
+- **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor.
+
+- **cs_pin** (*Optional*, :ref:`Pin Schema <config-pin_schema>`): The chip select pin.
+  Often marked ``T_CS`` on the board.
+
+- **irq_pin** (*Optional*, :ref:`Pin Schema <config-pin_schema>`): The touch detection pin.
+  Often marked ``T_IRQ`` on the board. If not specified the component will use polling
+  via SPI.
+
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
+  sensor. Defaults to ``50ms``. If ``irq_pin`` is specified the touch will be detected
+  nearly instantaneously and this setting will be used only for the release detection. 
+
+- **report_interval** (*Optional*, :ref:`config-time`): If not zero the component will
```

Comment:
config-time allows for ``never`` to be a valid value, it'd be better to accept that instead of zero to mean not report. Actually ``0`` is used as every loop update

Reply:
I added the `never` possibility to the code and made it default. It was actually only the update_interval that allowed for it, not config-time in general, so I added a similar way to handle it. I have updated the rest of the documentation based on your suggestions as well, thanks.